ts/features/mvl/screens/details/components/attachment/DownloadAttachmentConfirmationBottomSheet.tsx

```diff
@@ -202,22 +208,35 @@ export const useDownloadAttachmentConfirmationBottomSheet = (
 ) => {
   const { present, dismiss } = useIOBottomSheetRaw(BOTTOM_SHEET_HEIGHT);
   const dispatch = useIODispatch();
+  const { showAnimatedModal, hideModal } = useContext(LightModalContext);
 
   const openModalBox = () =>
     present(
       <DownloadAttachmentConfirmationBottomSheet
         onCancel={dismiss}
         onConfirm={({ dontAskAgain }) => {
           dispatch(mvlPreferencesSetWarningForAttachments(!dontAskAgain));
-          return handleDownloadResult(attachment, authHeader).then(() => {
-            dismiss();
-            if (options.showToastOnSuccess) {
-              showToast(
-                i18n.t("features.mvl.details.attachments.toast.success"),
-                "success"
-              );
-            }
-          });
+          return handleDownloadResult(
+            attachment,
+            authHeader,
+            (path, actionConfig) =>
```

Comment:
**⚠️ Opinionated comment incoming**

Since the `onConfirm` prop is becoming quite big, what do you think about separating _at least_ the `showPreview` callback of `handleDownloadResult` into a separate function? Something like:

```ts
const handleShowPreview = (path: string, actionConfig: ActionConfig) =>
    showAnimatedModal(
      <PdfPreview
        path={path}
        onClose={hideModal}
        onError={_error => {
          dismiss();
          showToast(
            i18n.t(
              "features.mvl.details.attachments.bottomSheet.failing.details"
            )
          );
        }}
        actionConfig={actionConfig}
      />,
      BottomTopAnimation
    );
```

Reply:
It's a fair point, yet given that this part will likely be reworked soon I'd rather keep it as it is.
I'll make sure to address this concern once we get the big picture 👍